lib/src/realm_object.dart

```diff
@@ -21,23 +21,64 @@ import 'realm_property.dart';
 
 abstract class RealmAccessor {
   T get<T>(RealmObject object, String name);
-  void set<T>(RealmObject object, String name, T value);
+  void set<T>(RealmObject object, String name, T value, [bool isDefault = false]);
+
+  static final Map<Type, Map<String, Object?>> _defaultValues = <Type, Map<String, Object?>>{};
```

Comment:
Why not let the factories registered on RealmObject handle default values?

Reply:
We discussed this on the round table, so resolving it